Core.Collectors/Web/HttpClientWrapper.cs

```diff
@@ -66,16 +77,24 @@ private async Task<HttpResponseMessage> MakeRequestAsync(string requestUrl, Http
             }
 
             request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            // Setup authentication
             string authenticationHeader = await authentication.GetAuthorizationHeaderAsync().ConfigureAwait(false);
             request.Headers.Authorization = new AuthenticationHeaderValue(authentication.Schema, authenticationHeader);
-            foreach (KeyValuePair<string, string> additionalHeader in authentication.AdditionalWebRequestHeaders)
+
+            // Setup additional headers
+            foreach (KeyValuePair<string, string> additionalHeader in authentication.AdditionalWebRequestHeaders.Union(additionalHeaders))
```

Comment:
Just make sure - is the `Union` intending to handle duplicate for Headers? As it won't because `KeyValuePair` are different when value is different so this essentially will `Concat` those altogether which I think is what you are intending but just confirm. (Eg if two pair `<HeaderFoo,Value1>` & `<HeaderFoo,Value2>` presents both will be added to request)

Reply:
In which case:
```suggestion
            foreach (KeyValuePair<string, string> additionalHeader in authentication.AdditionalWebRequestHeaders.Concat(additionalHeaders))
```
Because that's super cheap and doesn't confuse me when I see this again two days from now. :-)